README.md

```diff
@@ -39,193 +39,116 @@ do that in the [yagna repository](https://github.com/golemfactory/yagna) and in
+  API, the creation of a requestor agent should be straightforward and require minimal effort.
+  You can use examples contained in this repository as references, the directory
+  [`examples/hello-world/`](https://github.com/golemfactory/yapapi/tree/master/examples/hello-world)
+  contains minimal examples of fully functional requestor agents and
+  is therefore the best place to start exploring.
+
 
 ### Components
 
 There are a few components that are crucial for any requestor agent app:
 
-#### Executor
-
-The heart of the high-level API is the requestor's task executor (`yapapi.Executor`).
-You tell it, among others, which package (VM image) will be used to run your task,
-how much you'd like to pay and how many providers you'd like to involve in the execution.
-Finally, you feed it the worker script and a list of `Task` objects to execute on providers.
-
-#### Worker script
-
-The `worker` will most likely be the very core of your requestor app. You need to define
-this function in your agent code and then you pass it to the Executor.
-
-It receives a `WorkContext` (`yapapi.WorkContext`) object that serves
+#### Golem
+
+The heart of the high-level API is the `Golem` class (`yapapi.Golem`), which serves as 
+the "engine" of a requestor agent. `Golem` is responsible for finding providers interested in
+the jobs you want to execute, negotiating agreements with them and processing payments. It also
+implements core functionality required for sending commands to providers that 
+agreed to run your tasks.
+
+`Golem` provides two entry points for executing jobs on the Golem network, corresponding to the
+two basic modes of operation of a requestor agent:
+
+* The method `execute_tasks` allows you to submit a _task-based_ job for execution. Arguments
+to this method must include a sequence of independent _tasks_ (units of work) to be distributed
+among providers, a _payload_ (a VM image) required to compute them, and a _worker_ function, which
+will be used to convert each task to a sequence of steps to be executed on a provider. You may also
+specify the timeout for the whole job, the maximum number of providers used at any given time,
+and the maximum amount that you want to spend.
+
+* The method `run_service` allows you, as you probably guessed, to run a _service_ on Golem. 
+Instead of a task-processing worker function, an argument to `run_service` is a
+class (a subclass of `yapapi.Service`) that implements the behaviour of your service in various
+stages of its lifecycle (when it's _starting_, _running_ etc.). Additionally, you may specify
+the number of _service instances_ you want to run and the service expiration datetime.
+
+Prior to version `0.6.0`, only task-based jobs could be executed. They are probably conceptually
+simpler than services, and therefore in this `README` we focus on jobs of this type. 
+For more information on both types of jobs please refer to our [handbook](https://handbook.golem.network).
```

Comment:
seems redundant for me... maybe we should a short intro to Services instead? (in a separate PR perhaps?)

```suggestion
Prior to version `0.6.0`, only task-based jobs could be executed. For more information on both types of jobs please refer to our [handbook](https://handbook.golem.network).
```